source/org/zfin/database/repository/PostgresRepository.java

```diff
@@ -25,62 +17,29 @@ public class PostgresRepository {
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(PostgresSession.class);
-        if (formBean.getOrderBy() != null)
-            criteria.addOrder(Order.asc(formBean.getOrderBy()));
-        else
-            criteria.addOrder(Order.asc("dbname"));
-        return criteria.list();
+        String fromPostgresSession = "from PostgresSession order by dbname";
+        Query<PostgresSession> query = session.createQuery(fromPostgresSession, PostgresSession.class);
+        return query.list();
     }
 
     public static List<PostgresSession> getAllSessions(DatabaseFormBean formBean) {
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(PostgresSession.class);
-        criteria.add(Restrictions.eq("dbname", ZfinPropertiesEnum.DB_NAME.value()));
-
+        String hql = "from PostgresSession where dbname = :dbname";
         if (formBean.isActive())
-            criteria.add(Restrictions.eq("state", "active"));
-
-        criteria.addOrder(Order.desc("dateLastUsed"));
-        return (List<PostgresSession>) criteria.list();
-    }
-
-    private static List<SysSession> createSystemSessions(List<Object[]> list) {
-        List<SysSession> sessions = new ArrayList<SysSession>(list.size());
-        for (Object[] object : list) {
-            SysSession session = new SysSession();
-            session.setSid((Integer) object[0]);
-            session.setUserName((String) object[1]);
-            session.setUid((Short) object[2]);
-            session.setPid((Integer) object[3]);
-            session.setHostname((String) object[4]);
-            Date startDate = (Date) object[18];
-            session.setStartDate(toPstFromGMT(startDate));
-            SysOpenDb sysOpenDb = new SysOpenDb();
-            sysOpenDb.setName((String) object[21]);
-            sysOpenDb.setIsolation((Short) object[25]);
-            sysOpenDb.setSid((Integer) object[19]);
-
-            session.setSysOpenDb(sysOpenDb);
-            sessions.add(session);
-        }
-        return sessions;
-    }
+            hql += " AND state = state";
+        hql += " order by dateLastUsed";
+        Query<PostgresSession> query = session.createQuery(hql, PostgresSession.class);
+        query.setParameter("dbname", ZfinPropertiesEnum.DB_NAME.value());
+        if (formBean.isActive())
+            query.setParameter("state", "active");
 
-    public static Date toPstFromGMT(Date date) {
-        TimeZone pst = TimeZone.getTimeZone("PST");
-        return new Date(date.getTime() + pst.getRawOffset());
+        return query.list();
     }
 
-
     public static List<String> getAllDbNames() {
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(PostgresSession.class);
-        criteria.setProjection(Projections.distinct(Projections.property("dbname")));
```

Comment:
I'm not familiar with projections. I guess we don't need them here?

Reply:
I'll check this out in a future sprint. It's only used on a dev tools page.